src/cirrus/documentation_utils.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/env python
+"""
+_documentation_utils_
+
+utils for building and uploading Sphinx documentation
+
+"""
+import json
+import os
+import sys
+import tarfile
+
+from fabric.operations import local
+from requests_toolbelt import MultipartEncoder
+import pluggage.registry
+
+from cirrus.configuration import load_configuration
+from cirrus.logger import get_logger
+from cirrus.plugins.jenkins import JenkinsClient
+
+LOGGER = get_logger()
+
+
+def get_plugin(plugin_name):
+    """
+    _get_plugin_
+
+    Get the deploy plugin requested from the factory
+    """
+    factory = pluggage.registry.get_factory(
+        'upload',
+        load_modules=['cirrus.plugins.uploaders']
```

Comment:
So this factory is used by the pypi upload, right? I would suggest adding a new one for documentation uploaders, otherwise you could potentially end up trying to upload docs to pypi or artifacts to doc servers. Maybe add a plugin package called "publishers" and frame out support for readthedocs.org as well?